File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s826491904_commented.cpp
#include <iostream>
#include <algorithm>

int main() {
    int N;
    std::cin >> N;
    int max, min;
    std::cin >> max;
    min = max;
    for (int i = 1; i < N; i++) {
        int in;
        std::cin >> in;
        max = std::max(max, in);
        min = std::min(min, in);
    }
    std::cout << max - min << std::endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s826491904_commented.py
n = int(input())
first = int(input())
max_val = first
min_val = first
for _ in range(n-1):
    num = int(input())
    max_val = max(max_val, num)
    min_val = min(min_val, num)
print(max_val - min_val)
# End of Code
